### ISSUE TITLE
Add functions to check for capslock & numlock key states

### ISSUE THREAD

**Issue by ghost** [multitheftauto/mtasa-blue#1271]
**Is your feature request related to a problem? Please describe.**
We've currently no way of finding out wheter capslock or numlock are active or not. It could be useful for e.g show drawings when caps lock are active.

**Describe the solution you'd like**

Add the following functions (client only):
`bool isCapsLockEnabled()`
`bool isNumLockEnabled()`

**Additional context**
I wanted to add those functions but i'm unsure about naming - any ideas ?

**Comment by thisdp** [comment]
bool isScrollLockEnabled()

**Comment by qaisjp** [comment]
https://stackoverflow.com/q/13905342/1517394 -> https://wiki.multitheftauto.com/wiki/GetKeyState

from comments:

> Doesn't that just get the state of the key? Caps lock is usually used as a toggle. Does this really give you the toggle state for the whole machine? – user1994036 Dec 22 '14 at 16:54

> Yup, it really gives you the toggle state. It's not that intuitive, that this method gives you key presses by the user as well as overall key toggle states. – Jan Feldmann Aug 27 '16 at 9:38

additionally, from the second answer:

> Also works with VK_SCROLL for Scroll Lock and VK_NUMLOCK for Num Lock. – Jan Feldmann Aug 27 '16 at 9:41


haven't tested though.

**Comment by xLuxy** [comment]
> https://stackoverflow.com/q/13905342/1517394 -> https://wiki.multitheftauto.com/wiki/GetKeyState
> 
> from comments:
> 
> > Doesn't that just get the state of the key? Caps lock is usually used as a toggle. Does this really give you the toggle state for the whole machine? – user1994036 Dec 22 '14 at 16:54
> 
> > Yup, it really gives you the toggle state. It's not that intuitive, that this method gives you key presses by the user as well as overall key toggle states. – Jan Feldmann Aug 27 '16 at 9:38
> 
> additionally, from the second answer:
> 
> > Also works with VK_SCROLL for Scroll Lock and VK_NUMLOCK for Num Lock. – Jan Feldmann Aug 27 '16 at 9:41
> 
> haven't tested though.

@qaisjp i was talking about naming lua functions - ik that you could use `GetAsyncKeyState` / `GetkeyState`

**Comment by qaisjp** [comment]
I am suggesting that new functions are not necessary because it's possible that the existing Lua function [getKeyState](https://wiki.multitheftauto.com/wiki/GetKeyState) already works to check the caps lock state.

If that is the case, then we can extend getKeyState to support numlock and scroll lock.

**Comment by ghost** [comment]
Yeah i just deleted my comment. Gonna test and make an PR if it's not working

**Comment by thisdp** [comment]
getKeyState is good, but can not detect numlock,capslock and scrolllock currently.

**Comment by prnxdev** [comment]
Actually .. scroll lock and num lock is kinda weird idea because it doesn't change anything. Caps lock does because it changes lower case letters to upper case so you can show someone info, for example, near password input "watch out, you have caps lock on" or something like that.

**Comment by LosFaul** [comment]
i don't understand what's weird for you with numlock?
it makes for example 8, 4, 6, 2 behave like the arrow keys

**Comment by prnxdev** [comment]
But still you detect it as arrow keys, or not?
